test/models/sync_plan_test.rb

```diff
@@ -129,5 +129,25 @@ def test_schedule_format_disabled
       assert_match(/\/PT24H$/, schedule)
       assert_includes schedule, @plan.sync_date.iso8601
     end
+
+    def test_add_product
```

Comment:
I would rename these tests so that its clear that they are testing the `product_enabled` validation, rather than broader add/remove product tests.